test/unit/gateways/checkout_v2_test.rb

```diff
@@ -409,7 +409,8 @@ def test_successful_authorize_and_capture_with_3ds
           version: '1.0.2',
           eci: '05',
           cryptogram: '1234',
-          xid: '1234'
+          xid: '1234',
+          status: 'Y'
```

Comment:
Hi Bertrand,

Is the conditional check necessary in the unit tests? 
I would prefer avoiding introducing expects/with implementations in the current tests as they're not used at all at this stage.

Would replacing
`status: 'Y'`
with
`authentication_response_status: 'Y'`
work?

I noticed that "cryptogram" needs to be changed to "cavv" in these tests as well (lines 411 and 436), can you please confirm? 

On another note I just ran the remote tests locally and they all passed. 
Thanks

Reply:
Hello @nicolas-maalouf-cko!

Great news that the remote tests pass!

Here is my understanding.

These unit tests are lacking assertions/expectations even before your PR. For instance, `test_successful_authorize_and_capture_with_3ds` passes even when you comment `three_d_secure` altogether:
```ruby
  def test_successful_authorize_and_capture_with_3ds
    response = stub_comms do
      options = {
        execute_threed: true,
        attempt_n3d: true,
        # three_d_secure: {
        #   version: '1.0.2',
        #   eci: '05',
        #   cryptogram: '1234',
        #   xid: '1234'
        # }
      }
      @gateway.authorize(@amount, @credit_card, options)
    end.respond_with(successful_authorize_response)
 ```   
So I agree with you that `cryptogram ` is not correct because it is not one of the Standardized 3DS Fields [here](https://github.com/activemerchant/active_merchant/wiki/Standardized-3DS-Fields). However, changing it to `cavv` would not improve the test.

This PR is changing the behavior of the production code and needs tests for that. This PR should introduce tests that fail if
1. `status` is not set to `authentication_response_status` if `authentication_response_status` is present 
1. `status` is not set to `directory_response_status ` if `authentication_response_status` is absent and `directory_response_status ` is present 

I understand this PR does not introduce such tests.

What do you think?

Reply:
@nicolas-maalouf-cko 

eg
```ruby
  def test_provided_authentication_response_status_is_included_in_3ds_status
    response = stub_comms do
      options = {
        execute_threed: true,
        attempt_n3d: true,
        three_d_secure: {
          version: '1.0.2',
          eci: '05',
          cryptogram: '1234',
          xid: '1234',
          authentication_response_status: 'any_authentication_response_status'
        }
      }
      @gateway.authorize(@amount, @credit_card, options)
    end.check_request do |_endpoint, data, _headers|
      json_data = JSON.parse(data)
      assert_equal('any_authentication_response_status', json_data["3ds"]["status"])
    end.respond_with(successful_authorize_response)

    assert_success response
    assert_equal 'pay_fj3xswqe3emuxckocjx6td73ni', response.authorization
  end
```  

and more in order to cover the code change.

Reply:
@nicolas-maalouf-cko 

eg
```ruby
  def test_provided_authentication_response_status_is_included_in_3ds_status
    response = stub_comms do
      options = {
        execute_threed: true,
        attempt_n3d: true,
        three_d_secure: {
          version: '1.0.2',
          eci: '05',
          cryptogram: '1234',
          xid: '1234',
          authentication_response_status: 'any_authentication_response_status'
        }
      }
      @gateway.authorize(@amount, @credit_card, options)
    end.check_request do |_endpoint, data, _headers|
      json_data = JSON.parse(data)
      assert_equal('any_authentication_response_status', json_data["3ds"]["status"])
    end.respond_with(successful_authorize_response)

    assert_success response
  end
```  

and more in order to cover the code change.